File: gradle-plugin/ibc/ibc-api/src/main/java/com/jamesfchen/ibc/router/RoutersManager.java
package com.jamesfchen.ibc.router;

import static com.jamesfchen.ibc.Constants.ROUTER_TAG;

import android.util.Log;

import androidx.collection.ArrayMap;

/**
 * Copyright ® $ 2021
 * All right reserved.
 *
 * @author jamesfchen
 * @email hawksjamesf@gmail.com
 * @since 7月/08/2021  周四
 */
public class RoutersManager {
    private ArrayMap<String, ISchemaRouter> schemaRouters;
    private ArrayMap<String, IModuleRouter> moduleRouters;
    private ArrayMap<String, Class<?>> registerRouters;
    private static final String ROUTER_CONFIG = "BundleManifest.xml";
    RoutersManager(){
        moduleRouters = new ArrayMap<>();
        schemaRouters = new ArrayMap<>();
        registerRouters = new ArrayMap<>();
    }
    public static RoutersManager getInstance() {
        return LazyHolder.INSTANCE;
    }


    void getFlutterRouters() {
    }

    void getReactNativeRouters() {
    }

    void getNativeRouters() {
    }

    public void register(String routerName, Class<?> clz) {
        registerRouters.put(routerName, clz);
    }

    public IModuleRouter findModuleRouter(String routerName) {
        IModuleRouter moduleRouter = moduleRouters.get(routerName);
        if (moduleRouter ==null){
            Class<?> aClass = registerRouters.get(routerName);
            if (aClass ==null){
                throw new IllegalArgumentException(routerName +" 路由器没有注册");
            }
            try {
                moduleRouter = (IModuleRouter) aClass.newInstance();
                moduleRouters.put(routerName, (IModuleRouter) moduleRouter);
            } catch (IllegalAccessException | InstantiationException e) {
                e.printStackTrace();
                Log.d(ROUTER_TAG,Log.getStackTraceString(e));
                return null;
            }
        }
        return moduleRouter;
    }

    public ISchemaRouter findSchemaRouter(String routerName) {
        ISchemaRouter schemaRouter = schemaRouters.get(routerName);
        if (schemaRouter ==null){
            Class<?> aClass = registerRouters.get(routerName);
            if (aClass ==null){
                throw new IllegalArgumentException(routerName +" 路由器没有注册");
            }
            try {
                schemaRouter = (ISchemaRouter) aClass.newInstance();
                schemaRouters.put(routerName, (ISchemaRouter) schemaRouter);
            } catch (IllegalAccessException | InstantiationException e) {
                e.printStackTrace();
                Log.d(ROUTER_TAG,Log.getStackTraceString(e));
                return null;
            }
        }
        return schemaRouter;
    }
    private static class LazyHolder {
        static final RoutersManager INSTANCE = new RoutersManager();

    }
}

File: android/framework/ibc/src/main/java/com/jamesfchen/ibc/route/RoutersManager.kt
package com.jamesfchen.ibc.route

import android.content.Context
import androidx.collection.ArrayMap

/**
 * Copyright ® $ 2021
 * All right reserved.
 *
 * @author: jamesfchen
 * @email: hawksjamesf@gmail.com
 * @since: 六月/30/2021  星期三
 */
object RoutersManager {
    private var hybridRouters : ArrayMap<String, IHybridRouter>? = null
    private var flutterRouters :  ArrayMap<String, IFlutterRouter>? = null
    private var reactNativeRouters :  ArrayMap<String, IReactNativeRouter>? = null
    private var nativeRouters :  ArrayMap<String, INativeRouter>? = null
    private const val ROUTER_CONFIG="BundleManifest.xml"

    fun getHybridRouters(cxt:Context):ArrayMap<String, IHybridRouter>{
        if (hybridRouters ==null){
//            XmlParser.parse(cxt.assets.open(ROUTER_CONFIG))
        }
        return hybridRouters!!
    }
    fun getFlutterRouters(){}
    fun getReactNativeRouters(){}
    fun getNativeRouters(){}
}